tests/com_wrapper_ut/com_wrapper_ut.c

```diff
@@ -174,6 +193,35 @@ TEST_FUNCTION(when_malloc_fails_create_fails)
     test_object_destroy(test_object);
 }
 
+/* Tests_SRS_COM_WRAPPER_66_001: [ DEFINE_COM_WRAPPER_OBJECT_WITH_MALLOC_FUNCTIONS shall generate constructors for the COM object for each implemented interface. ]*/
+/* Tests_SRS_COM_WRAPPER_66_002: [ DEFINE_COM_WRAPPER_OBJECT_WITH_MALLOC_FUNCTIONS shall generate all the underlying Vtbl structures needed for the COM object. ]*/
+/* Tests_SRS_COM_WRAPPER_66_003: [ DEFINE_COM_WRAPPER_OBJECT_WITH_MALLOC_FUNCTIONS shall use malloc_func to allocate memory for the COM wrapper object. ]*/
+/* Tests_SRS_COM_WRAPPER_66_004: [ DEFINE_COM_WRAPPER_OBJECT_WITH_MALLOC_FUNCTIONS shall use free_func to free the memory associated with the COM wrapper object. ]*/
+TEST_FUNCTION(create_with_custom_allocator_uses_provided_functions)
+{
+    // arrange
+    TEST_OBJECT_HANDLE test_object = test_object_create("haga");
+    umock_c_reset_all_calls();
+
+    STRICT_EXPECTED_CALL(ut_custom_malloc(sizeof(TEST_OBJECT_UT_ALLOC_HANDLE_COM_WRAPPER)));
+
+    // act
+    ITestInterface* custom_interface = COM_WRAPPER_CREATE(TEST_OBJECT_UT_ALLOC_HANDLE, ITestInterface, (TEST_OBJECT_UT_ALLOC_HANDLE)test_object, test_object_destroy);
+
+    // assert
+    ASSERT_IS_NOT_NULL(custom_interface);
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+
+    // verify free is called on release
+    umock_c_reset_all_calls();
+    STRICT_EXPECTED_CALL(test_object_destroy(test_object));
+    STRICT_EXPECTED_CALL(ut_custom_free(IGNORED_ARG));
+
+    (void)custom_interface->lpVtbl->Release(custom_interface);
```

Comment:
There should be a separate test for testing Release under the Release section of tests right after Release_decrements_the_reference_count_to_zero_and_calls_handle_destroy